slang-checks/src/main/java/com/sonarsource/slang/checks/IfConditionalAlwaysTrueOrFalseCheck.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.IfTree;
+import com.sonarsource.slang.api.LiteralTree;
+import com.sonarsource.slang.api.Tree;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import java.util.Arrays;
+import java.util.List;
+import org.sonar.check.Rule;
+
+@Rule(key = "S1145")
+public class IfConditionalAlwaysTrueOrFalseCheck implements SlangCheck {
+  private static final List<String> BOOLEAN_LITERALS = Arrays.asList("true", "false");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(IfTree.class, (ctx, ifTree) -> {
+      Tree condition = ifTree.condition();
+      if (condition instanceof LiteralTree && BOOLEAN_LITERALS.contains(((LiteralTree) condition).value())) {
```

Comment:
I think we should cover also the cases
- `if (!true)` and `if(!false)`
- And expression that contains a `false` literal
- Or expression that contains a `true` literal